src/plugins/preauth/pkinit/pkinit_crypto_openssl.c

```diff
@@ -4287,28 +4286,26 @@ pkinit_get_certs_pkcs12(krb5_context context,
     EVP_PKEY *y = NULL;
 
     if (idopts->cert_filename == NULL) {
-        pkiDebug("%s: failed to get user's cert location\n", __FUNCTION__);
+        TRACE_PKINIT_NO_CERT(context);
```

Comment:
Looking at the code, I think this and some of the similar trace messages are unreachable.  pkinit_get_certs_pkcs12() is only called from crypto_load_certs() when idopts->idtype is IDTYPE_PKCS12, and in that case pkinit_identity.c used parse_pkcs12_options() and set cert_filename and key_filename.

There's an opportunity for code cleanup here (perhaps eliminating the pkinit_identity_opts type and using separate entry points into the crypto file for different ID types), but in the meantime I don't think we want to add tracing calls that can't be triggered.